src/ethereum-json-rpc-client/src/state_machine_tests_client.rs

```diff
@@ -0,0 +1,102 @@
+use std::collections::HashMap;
+use std::future::Future;
+use std::pin::Pin;
+
+use anyhow::Context;
+use candid::{CandidType, Deserialize};
+use ic_canister_client::StateMachineCanisterClient;
+use jsonrpc_core::{Request, Response};
+use serde::Serialize;
+use serde_bytes::ByteBuf;
+
+use crate::Client;
+
+impl Client for StateMachineCanisterClient {
```

Comment:
The `Client` trait cannot be implemented for the generic `IcCanisterClient` because the latter is not Sync.